### PR TITLE
Added support for generators on publish

### LINKED ISSUE #17: Support generators on publish
`Ensign.publish()` accepts an iterable of events. However we could also allow the user to pass in a generator which yields the events.

For instance:

```python
def event_source():
    ...
    yield event

ensign = Ensign()
await ensign.publish(event_source())
```

### PR DESCRIPTION
**Pull request by vibhakar025**
Closes #17 

A new method has been implemented that accepts a single event from the generator and publishes it.

### CODE REVIEW

File: tests/pyensign/test_ensign.py [hunk 1]
```diff
@@ -102,6 +102,39 @@ async def test_publish_exists(self, mock_publish, mock_list, ensign):
 
         errors = await ensign.publish(name, events)
         assert len(errors) == 1
+    
+    @pytest.mark.asyncio
+    @patch("pyensign.connection.Client.list_topics")
+    @patch("pyensign.connection.Client.publish")
+    async def test_publish_with_generator(self, mock_publish, mock_list, ensign):
```
Comment (pdeziel) [inline tests/pyensign/test_ensign.py:109]: Thank you for adding this test, this is definitely a pattern I would expect users to use.

**Review by pdeziel — COMMENTED** [review]
@vibhakar025 Thank you for adding the test! To clarify the requirements in the issue, I was thinking we would just want to modify the original `publish` method to accept a generator (e.g. as a keyword argument).

However the test you added is certainly a valid use of the API, so I want to make sure that works. Can you run the test against the original method and see if it passes? If it works then I would consider this issue fixed (because we have a verifiable test case). If it doesn't, then we should fix the original method.

**Comment by pdeziel** [comment]
@vibhakar025 Just checking in to see how the changes are going. Do you have any questions I can answer for you? Let me know if you plan on continuing with this, otherwise I'd be happy to pick up where you left off and get this merged!

**Comment by vibhakar025** [comment]
Hi @pdeziel, apologies, I was away. I am trying to understand the project again and setting it up from scratch. I followed the steps provided but unable to get the tests running successfully. Getting this error - 
```
event = event_pb2.Event(
            user_defined_id="1",
        )
   ValueError: Protocol message Event has no "user_defined_id" field.
```
Am i missing something here?

**Comment by pdeziel** [comment]
> 

Hi @vibhakar025 , I noticed this today as well. There was a recent change to the protocol buffers [here](https://github.com/rotationalio/ensign/pull/502) so the `user_defined_id` field is no longer in that message. I have an open [PR](https://github.com/rotationalio/pyensign/pull/26) which fixes the tests. That should be merged tomorrow, sorry for the inconvenience!

In the meantime you should be able to remove the `user_defined_id` fields like I did in that PR and the tests should pass.

**Pull request closed without merging**